src/main/java/bio/terra/common/stairway/StairwayComponent.java

```diff
@@ -0,0 +1,138 @@
+package bio.terra.common.stairway;
+
+import static com.google.cloud.ServiceOptions.getDefaultProjectId;
+
+import bio.terra.common.kubernetes.KubeService;
+import bio.terra.common.kubernetes.KubernetesComponent;
+import bio.terra.stairway.Stairway;
+import bio.terra.stairway.exception.StairwayException;
+import bio.terra.stairway.exception.StairwayExecutionException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+
+/** A Spring Component for exposing an initialized {@link Stairway}. */
+@Component
+public class StairwayComponent {
+  private final Logger logger = LoggerFactory.getLogger(StairwayComponent.class);
+
+  private final StairwayConfiguration stairwayConfiguration;
+  private final StairwayJdbcConfiguration stairwayJdbcConfiguration;
+  private final Stairway stairway;
+  private final KubeService kubeService;
+
+  public enum Status {
+    INITIALIZING,
+    OK,
+    ERROR,
+    SHUTDOWN,
+  }
+
+  private final AtomicReference<Status> status = new AtomicReference<>(Status.INITIALIZING);
+
+  @Autowired
+  public StairwayComponent(
+      ApplicationContext applicationContext,
+      StairwayConfiguration stairwayConfiguration,
+      StairwayJdbcConfiguration stairwayJdbcConfiguration,
+      KubernetesComponent kubernetesComponent) {
+    this.stairwayConfiguration = stairwayConfiguration;
+    this.stairwayJdbcConfiguration = stairwayJdbcConfiguration;
+    this.kubeService = kubernetesComponent.get();
+    String stairwayClusterName = kubeService.getNamespace() + "buffer--stairwaycluster";
+    logger.info(
+        "Creating Stairway: name: [{}]  cluster name: [{}]",
+        kubeService.getPodName(),
+        stairwayClusterName);
+
+    // TODO(PF-314): Cleanup old flightlogs.
+    // TODO(PF-456):
+    Stairway.Builder builder =
+        Stairway.newBuilder()
+            .maxParallelFlights(stairwayConfiguration.getMaxParallelFlights())
+            .applicationContext(applicationContext)
+            .keepFlightLog(true)
+            .stairwayName(kubeService.getPodName())
+            .stairwayClusterName(stairwayClusterName)
+            .workQueueProjectId(getDefaultProjectId())
+            .enableWorkQueue(kubernetesComponent.isInKubernetes())
+            .stairwayHook(new TracingHook());
+    try {
+      stairway = builder.build();
+    } catch (StairwayExecutionException e) {
+      throw new IllegalArgumentException("Failed to build Stairway.", e);
```

Comment:
@yonghaoy ^

Reply:
This is certainly not an illegal argument. I think this is here because StairwayExecutionException was turned into a checked exception. We should probably undo using checked exceptions from Stairway. It has turned out to be a real PITA.